src/index.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { PathExt } from '@jupyterlab/coreutils';
+import { IFileBrowserFactory } from '@jupyterlab/filebrowser';
+import { GithubPuller } from './githubpuller';
+
+const assignmentListExtension: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlite/litegitpuller.plugin',
+  autoStart: true,
+  requires: [IFileBrowserFactory],
+  activate: (app: JupyterFrontEnd, browserFactory: IFileBrowserFactory) => {
+    console.log(
+      'JupyterLab extension @jupyterlite/litegitpuller is activated!'
+    );
+    if (!(app.name === 'JupyterLite')) {
```

Comment:
Should we let the plugin activate itself even if not in JupyterLite?

Since the app name can be configured this check might not be robust enough.

Reply:
It may conflict with `nbgitpuller` if both are installed in Jupyterlab.
I'll open an issue to fix the check on the current app.